### PR TITLE
.*:  replace `grpc.Dial` with `grpc.NewClient` in test files

### PR DESCRIPTION
**Pull request by hanut19**
Partially address: #7049 

Update tests where only direct replacement is enough to replace `grpc.Dial` with `grpc.NewClient` in test files

RELEASE NOTES: None

### CODE REVIEW

**Comment by purnesh42H** [comment]
@hanut19 please complete your CLA Authorization. Also, please ensure following the convention for PR description and title for next PRs

File: xds/internal/balancer/clusterresolver/e2e_test/aggregate_cluster_test.go
Comment (purnesh42H) [inline xds/internal/balancer/clusterresolver/e2e_test/aggregate_cluster_test.go:1187]: `failed to create new client for local test server`. Please apply this everywhere

Comment (hanut19) [reply xds/internal/balancer/clusterresolver/e2e_test/aggregate_cluster_test.go]: done please check

**Review by purnesh42H — APPROVED** [review]
lgtm

**Comment by purnesh42H** [comment]
@dfawley for second review

**Comment by dfawley** [comment]
Note that the CLAs are currently broken, so we can't merge this.

File: balancer/endpointsharding/endpointsharding_test.go
Comment (dfawley) [inline balancer/endpointsharding/endpointsharding_test.go:147]: "dial" is a verb.  "NewClient" is not.  Please fix the errors that don't have proper grammar.

Comment (purnesh42H) [reply balancer/endpointsharding/endpointsharding_test.go:147]: @hanut19 `failed to create new client`. Similar to what you changed elsewhere. I think you missed this one

**Comment by hanut19** [comment]
another PR is raised https://github.com/grpc/grpc-go/pull/7640

**Pull request closed without merging**
